### PR TITLE
Remove all references to habana_quantization_toolkit for 1.18

### PR DESCRIPTION
**Pull request by tthakkal**
# What does this PR do?
Removing all references to habana_quantization_toolkit  as it is removed in latest 1.18 release

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: server/text_generation_server/models/causal_lm.py [hunk 1]
```diff
@@ -796,20 +793,6 @@ def get_rope_scaling(self) -> Optional[Dict]:
...
-        return model
-
-    def prepare_model_for_quantization(self, model):
-        if hq_env.is_quantization_enabled:
-            if model.config.model_type == "llama":
-                self.patch_scoped_linear_all_reduce(model)
-            model = hq_env.prepare_model_for_quantization(model)
-        return model
-
     def patch_scoped_linear_all_reduce(self, model):
```
Comment (schoi-habana) [inline server/text_generation_server/models/causal_lm.py]: this has to be removed as it is copied to habana_quantization_env.py?

File: server/text_generation_server/models/vlm_causal_lm.py [hunk 2]
```diff
@@ -703,27 +700,6 @@ def get_rope_scaling(self) -> Optional[Dict]:
...
-            habana_quantization_toolkit.prep_model(model)
-        return model
-
-    def finish_quantization_measurements(self, model):
-        if hq_env.is_quantization_enabled:
-            import habana_quantization_toolkit
-            habana_quantization_toolkit.finish_measurements(self.model)
-        return model
-
     def patch_scoped_linear_all_reduce(self, model):
```
Comment (schoi-habana) [inline server/text_generation_server/models/vlm_causal_lm.py]: this has to be removed as it is copied to habana_quantization_env.py?

File: server/text_generation_server/habana_quantization_env.py [hunk 3]
```diff
@@ -17,11 +18,27 @@
...
     if is_quantization_enabled:
-        if os.getenv("USE_INC", "1") != "0":
-            from neural_compressor.torch.quantization import FP8Config, convert
-            config = FP8Config.from_json_file(quant_config)
-            model = convert(model, config)
-        else:
-            import habana_quantization_toolkit
-            habana_quantization_toolkit.prep_model(model)
+        htorch.core.hpu_set_env()
+        if model.config.model_type == "llama":
```
Comment (mandy-li) [inline server/text_generation_server/habana_quantization_env.py]: Only llama model needs to do this? @schoi-habana , i remember Falcon did it too, right?

**Review by mandy-li — COMMENTED** [review]
@tthakkal , pls see our comments in code

File: server/text_generation_server/models/causal_lm.py (hunk 1 above)
Comment (tthakkal) [reply server/text_generation_server/models/causal_lm.py]: removed

File: server/text_generation_server/models/vlm_causal_lm.py (hunk 2 above)
Comment (tthakkal) [reply server/text_generation_server/models/vlm_causal_lm.py]: removed, thanks for catching

File: server/text_generation_server/habana_quantization_env.py (hunk 3 above)
Comment (schoi-habana) [reply server/text_generation_server/habana_quantization_env.py]: yes OH does this for this series of models

https://github.com/huggingface/optimum-habana/blob/8d79a834c9571359745a491a686e702d28509ee1/examples/text-generation/utils.py#L424

Comment (tthakkal) [reply server/text_generation_server/habana_quantization_env.py]: @mandy-li I haven't changed this code, just consolidated code into one place. 
Do we need to add all those list models here ? that will be a change from existing code.

Comment (mandy-li) [reply server/text_generation_server/habana_quantization_env.py]: yes we should

Comment (tthakkal) [reply server/text_generation_server/habana_quantization_env.py]: added

**Review by mandy-li — APPROVED** [review]
lgtm

**Comment by mandy-li** [comment]
@regisss , pls review and merge if ok for you

**Review by regisss — APPROVED** [review]
LGTM!

**Pull request merged**